tiled/_tests/test_access_control.py

```diff
@@ -2,18 +2,66 @@
 
 import numpy
 import pytest
-from starlette.status import HTTP_403_FORBIDDEN
+from fastapi import HTTPException
+from starlette.status import HTTP_403_FORBIDDEN, HTTP_404_NOT_FOUND
 
+from ..access_policies import (
+    ALL_SCOPES,
+    PUBLIC_SCOPES,
+    SimpleAccessPolicy,
+    SpecialUsers,
+)
 from ..adapters.array import ArrayAdapter
 from ..adapters.mapping import MapAdapter
 from ..client import Context, from_context
+from ..client.utils import ClientError
 from ..server.app import build_app_from_config
+from ..server.core import NoEntry
 from .utils import enter_username_password, fail_with_status_code
 
 arr = numpy.ones((5, 5))
+arr_zeros = numpy.zeros((5, 5))
 arr_ad = ArrayAdapter.from_array(arr)
 
 
+class EntryBasedAccessPolicy(SimpleAccessPolicy):
```

Comment:
A docsting here would be useful:

> This is access policy demonstrates how the metadata on some nested child node can be efficiently consulted and incorporated in logic that determines access scopes. In this test example, the metadata on the node quite literally lists some scopes that it should not allow. In realistic examples it could be incorporated in site-specific logic.